FreeRTOS/Test/Target/tests/smp/multiple_tasks_running/multiple_tasks_running.c

```diff
@@ -0,0 +1,147 @@
+ *
+ * https://www.FreeRTOS.org
+ * https://github.com/FreeRTOS
+ *
+ */
+
+/**
+ * @file multiple_tasks_running.c
+ * @brief The user shall be able to schedule tasks across multiple identical processor cores
+ *        with one instance of FreeRTOS scheduler.
+ *
+ * Procedure:
+ *   - Create ( num of cores - 1 ) tasks and keep them in busy loop.
+ * Expected:
+ *   - All tasks are in running state.
+ */
+
+/* Kernel includes. */
+#include "FreeRTOS.h" /* Must come first. */
+#include "task.h"     /* RTOS task related API prototypes. */
+
+#include "unity.h"    /* unit testing support functions */
+/*-----------------------------------------------------------*/
+
+#if ( configNUMBER_OF_CORES < 2 )
+    #error This test is for FreeRTOS SMP and therefore, requires at least 2 cores.
+#endif /* if configNUMBER_OF_CORES != 2 */
+
+#if configRUN_MULTIPLE_PRIORITIES != 1
+    #error test_config.h must be included at the end of FreeRTOSConfig.h.
+#endif
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Function that implements a never blocking FreeRTOS task.
+ */
+static void prvEverRunningTask( void * pvParameters );
+
+/**
+ * @brief Test case "Multiple Tasks Running".
+ */
+static void Test_MultipleTasksRunning( void );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Handles of the tasks created in this test.
+ */
+static TaskHandle_t xTaskHanldes[ configNUMBER_OF_CORES - 1 ];
+/*-----------------------------------------------------------*/
+
+static void Test_MultipleTasksRunning( void )
+{
+    int i;
+    eTaskState xTaskState;
+
+    /* Delay for other cores to run tasks. */
+    vTaskDelay( pdMS_TO_TICKS( 10 ) );
+
+    /* Ensure that all the tasks are running. */
+    for( i = 0; i < configNUMBER_OF_CORES - 1; i++ )
+    {
+        xTaskState = eTaskGetState( xTaskHanldes[ i ] );
+
+        TEST_ASSERT_EQUAL_MESSAGE( eRunning, xTaskState, "Task is not running." );
+    }
+}
+/*-----------------------------------------------------------*/
+
+static void prvEverRunningTask( void * pvParameters )
+{
+    /* Silence warnings about unused parameters. */
+    ( void ) pvParameters;
+
+    for( ; ; )
+    {
+        /* Always running, put asm here to avoid optimization by compiler. */
+        __asm volatile ( "nop" );
+    }
+}
+/*-----------------------------------------------------------*/
+
+/* Runs before every test, put init calls here. */
+void setUp( void )
+{
+    int i;
+    BaseType_t xTaskCreationResult;
+
+    /* Create configNUMBER_OF_CORES - 1 low priority tasks. */
+    for( i = 0; i < configNUMBER_OF_CORES - 1; i++ )
+    {
+        xTaskCreationResult = xTaskCreate( prvEverRunningTask,
+                                           "EverRunning",
+                                           configMINIMAL_STACK_SIZE,
+                                           NULL,
+                                           configMAX_PRIORITIES - 2,
```

Comment:
Ensure configMAX_PRIORITIES > 2. Otherwise, idle task may be selected to run.

Reply:
Added at line 53